### ISSUE TITLE
TODO: Add title

### ISSUE THREAD

**Issue by matthewpersico** [koalaman/shellcheck#1170]
#### For bugs
- Rule Id (if any, e.g. SC1000): Any rule actually
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!bash

datafile='the/foo/file'

for i in $(grep -v '#' "$datafile")
do
    echo "$i"
done

#shellcheck disable=SC2013
for i in $(grep -v '#' "$datafile")
do
    echo "$i"
done
    
#shellcheck disable=SC2013
while read -r i
do
    echo "$i"
done < "$datafile"
    
```

#### Here's what shellcheck currently says:
```
Line 5	SC2013: To read lines rather than words, pipe/redirect to a 'while read' loop.
```
#### Here's what I wanted or expected to see:

TODO: Describe expected/desired output
```
Line 5	SC2013: To read lines rather than words, pipe/redirect to a 'while read' loop.
Line 5	SC2013: Does not apply to next line; remove it.

```
Sometimes when you go back and edit files, you manage to undo the issue that shellcheck was complaining about. That means the disable directive is no longer applicable and should be removed. Can we have the parser check for disabled checks and warn us if the disable no longer applies?
